### DISCUSSION TITLE
external processes for compilations

### DISCUSSION THREAD

**Discussion by ousia** [orbitalquark/textadept#216] [Q&A] [answered]
I compile my TeX sources with:

```lua
textadept.run.compile_commands.tex = 'context --purgeall "%f"'
```

In some cases, Textadept gets stuck due to faulty code.

If I try to kill the process with `top`, I only have `textadept` to kill.

With Geany, the `luametatex` process is independent from the editor.

This avoids some editor freezing and also enables killing the wrong external process.

How can I have external processes when compiling in Textadept?

Many thanks for your help.

**Comment by rgieseke** [comment]
What is shown when you run the command for a faulty file in a terminal? Does it just hang, waiting for some input?
If context finishes with some error code there should be no problem. Maybe there is an option to not stop at an error?

**Reply by rgieseke** [reply]
Something like halt-on-error but for Context?

https://tex.stackexchange.com/questions/258814/what-is-the-difference-between-interaction-nonstopmode-and-halt-on-error

**Comment by orbitalquark** [comment, marked as answer]
Textadept spawns processes as external child processes, but it's possible that the children don't show up in `top`. It's on my TODO list to come up with some sort of way to kill misbehaving processes via the GUI. Sorry about that :(

**Comment by orbitalquark** [comment]
Oh, I forgot about Tools > Stop. Does that work for you?

**Reply by ousia** [reply]
I didn’t know that such an option was available. Now I need another external process to hijack Textadept.

In some cases, Textadept becomes unresponsive (so I wonder whether stopping the compilation would even be detected by Textadept).

**Reply by ousia** [reply]
Today I had to stop compilation. Textadept became unresponsive for some seconds, but at the end it stopped compilation.

**Comment by paaguti** [comment]
I have another (hopefully) portable approach: a script that starts latexmk
and opens the .tex file in TextAdept (or any other editor) as child
processes. Once you modify anything in the .tex file and save, latexmk
kicks in the compiler and launches the PDF viewer on success.

On Linux, I have tested it on a smaller machine with TextAdept and
epdfviewer and on my main machine with Emacs and evince. On macOS, I have
tested both TextAdept and Emacs collaborating with Skim. Additionally, I
also have forward and backward search between the PDF and the .tex on all
combinations.

Hope it helps...


On Tue, 21 Jun 2022 at 18:34, Pablo Rodríguez ***@***.***>
wrote:

> I didn’t know that such an option was available. Now I need another
> external process to hijack Textadept.
>
> In some cases, Textadept becomes unresponsive (so I wonder whether
> stopping the compilation would even be detected by Textadept).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orbitalquark/textadept/discussions/216#discussioncomment-2995644>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAZKU4A54G3VASPEHANIWGDVQHVJZANCNFSM5ZE6ATPQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
Fragen sind nicht da um beantwortet zu werden,
Fragen sind da um gestellt zu werden
Georg Kreisler

Headaches with a Juju log:
unit-basic-16: 09:17:36 WARNING juju.worker.uniter.operation we should run
a leader-deposed hook here, but we can't yet

**Reply by ousia** [reply]
Many thanks for your reply, @paaguti.

Not sure is the thing I need (in any case, I don’t want to compile each time I save sources).

Sorry, but it seems an unnecessarily complex processing to me.
